tests/unit/contentproviders/test_hydroshare.py

```diff
@@ -0,0 +1,212 @@
+            ],
+        ):
+            with patch.object(
+                Hydroshare, "_urlretrieve", side_effect=[(hydro_path, None)]
+            ):
+                hydro = Hydroshare()
+                hydro.resource_id = "b8f6eae9d89241cf8b5904033460af61"
+                spec = {
+                    "host": {
+                        "hostname": [
+                            "https://www.hydroshare.org/resource/",
+                            "http://www.hydroshare.org/resource/",
+                        ],
+                        "django_irods": "https://www.hydroshare.org/django_irods/download/bags/",
+                    },
+                    "resource": "123456789",
+                }
+
+                with TemporaryDirectory() as d:
+                    output = []
+                    for l in hydro.fetch(spec, d):
+                        output.append(l)
+
+                    unpacked_files = set(os.listdir(d))
+                    expected = set(["some-other-file.txt", "some-file.txt"])
+                    assert expected == unpacked_files
+
+
+def test_fetch_bag_failure():
+    with hydroshare_archive() as hydro_path:
+        with patch.object(
+            Hydroshare, "urlopen", side_effect=[MockResponse("application/html", 500)]
+        ):
+            hydro = Hydroshare()
+            spec = {
+                "host": {
+                    "hostname": [
+                        "https://www.hydroshare.org/resource/",
+                        "http://www.hydroshare.org/resource/",
+                    ],
+                    "django_irods": "https://www.hydroshare.org/django_irods/download/bags/",
+                },
+                "resource": "123456789",
+            }
+            with TemporaryDirectory() as d:
+                output = []
+                try:
+                    for l in hydro.fetch(spec, d):
+                        output.append(l)
+                    print("ContentProviderException should have been thrown")
+                    assert False
+                except ContentProviderException:
+                    assert "Failed to download bag. status code 500.\n" == output[-1]
+
+
+def test_fetch_bag_timeout():
+    with hydroshare_archive() as hydro_path:
+        with patch.object(
+            Hydroshare, "urlopen", side_effect=[MockResponse("application/html", 200)]
+        ):
+            hydro = Hydroshare()
+            spec = {
+                "host": {
+                    "hostname": [
+                        "https://www.hydroshare.org/resource/",
+                        "http://www.hydroshare.org/resource/",
+                    ],
+                    "django_irods": "https://www.hydroshare.org/django_irods/download/bags/",
+                },
+                "resource": "123456789",
+            }
+            with TemporaryDirectory() as d:
+                output = []
+                try:
+                    for l in hydro.fetch(spec, d, timeout=0):
```

Comment:
Instead of `try-except`ing yourself to find out if an exception was raised can you use `pytest.raises` as context manager here?

```python
import pytest


def myfunc():
    raise ValueError("Exception 123 raised")


def test_match():
    with pytest.raises(ValueError, match=r".* 123 .*"):
        myfunc()
```